ecosystem/gstreamer_plugin/gst_mtl_common.c

```diff
@@ -422,5 +456,50 @@ mtl_handle gst_mtl_common_init_handle(struct mtl_init_params* p, StDevArgs* devA
   }
   *log_level = mtl_init_params.log_level;
 
-  return mtl_init(&mtl_init_params);
+  if (force_to_initialize_new_instance) {
+    GST_INFO("MTL shared handle ignored");
+    pthread_mutex_unlock(&common_handle.mutex);
+
+    return mtl_init(&mtl_init_params);
```

Comment:
I wonder if mtl_init is thread safe.  I'm not sure, but I believe it is not. Just to be sure, please initialize it under lock.

Reply:
Done